source/extensions/clusters/common/logical_host.h

```diff
@@ -96,6 +88,8 @@ class RealHostDescription : public HostDescription {
     return logical_host_->canCreateConnection(priority);
   }
   HealthCheckHostMonitor& healthChecker() const override { return logical_host_->healthChecker(); }
+  void setHealthChecker(HealthCheckHostMonitorPtr&&) override {}
+  void setOutlierDetector(Outlier::DetectorHostMonitorPtr&&) override {}
```

Comment:
The root cause is, I think, because the HostDescription and Host abstractions are not quite right and I think there should not be an inheritance. Perhaps the abstractions were originally created before there was forward-proxy functionality.

Thus we need this delegate implementation for a Host which reflects flexibility and overrides for some of the fields and not other ones.

From a C++ perspective this manifests in a delegation tactic that uses a HostConstSharedPtr, so all the setters have to be no-ops.

I can add some comments.